src/Microsoft.Azure.SignalR.Protocols/Microsoft.Azure.SignalR.Protocols.csproj

```diff
@@ -1,21 +1,48 @@
-<Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
-    <Description>.NET Standard SDK for Azure SignalR Service protocol.</Description>
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <RootNamespace>Microsoft.Azure.SignalR.Protocol</RootNamespace>
-    <GenerateDocumentationFile>true</GenerateDocumentationFile>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <Compile Include="..\Common\BinaryMessageFormatter.cs" Link="Internal\BinaryMessageFormatter.cs" />
-    <Compile Include="..\Common\BinaryMessageParser.cs" Link="Internal\BinaryMessageParser.cs" />
-    <Compile Include="..\Common\MemoryBufferWriter.cs" Link="Internal\MemoryBufferWriter.cs" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <PackageReference Include="MessagePack" Version="$(MessagePackPackageVersion)" />
-    <PackageReference Include="Microsoft.Extensions.Primitives" Version="$(MicrosoftExtensionsPrimitivesPackageVersion)" />
-    <PackageReference Include="System.Buffers" Version="$(SystemBuffersPackageVersion)" />
-    <PackageReference Include="System.Memory" Version="$(SystemMemoryPackageVersion)" />
-  </ItemGroup>
-</Project>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>.NET Standard SDK for Azure SignalR Service protocol.</Description>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <RootNamespace>Microsoft.Azure.SignalR.Protocol</RootNamespace>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Common\BinaryMessageFormatter.cs" Link="Internal\BinaryMessageFormatter.cs" />
+    <Compile Include="..\Common\BinaryMessageParser.cs" Link="Internal\BinaryMessageParser.cs" />
+    <Compile Include="..\Common\MemoryBufferWriter.cs" Link="Internal\MemoryBufferWriter.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Primitives" Version="$(MicrosoftExtensionsPrimitivesPackageVersion)" />
+    <PackageReference Include="System.Buffers" Version="$(SystemBuffersPackageVersion)" />
+    <PackageReference Include="System.Memory" Version="$(SystemMemoryPackageVersion)" />
+
+    <PackageReference Include="Microsoft.VisualStudio.Validation" Version="$(MicrosoftVisualStudioValidationPackageVersion)" />

```

Comment:
`<PrivateAssets>all</PrivateAssets>` to prevent the reference from exposing?

Reply:
Given this requires a runtime assembly, `PrivateAssets=all` would generate runtime failures. Use `PrivateAssets=compile` instead so that others don't see it as a reference assembly but it still ships with the app.

Reply:
humm ok i though it was kinda like a `Debug.Assert` so that it would not be built/output in `Release` configuration.

the  2 questions, what the general idea aside the if check > fail
do i still needs a `PrivateAsset` then if it's useful at runtime ? is it a good/bad idea to let this dependency be visible as a transitive ? (probably not important i guess then ?

i'll have to reopen the docs on PrivateAssets to get a better understanding at it ^^
especially ... what if it's set on "Compile" and some use it too (yeah i have a weird mind i try to break everything before it even work)

Reply:
`PrivateAssets="compile"` prevents your consumers from seeing the types inside it. They can still add a package ref to it themselves and to reference it if they want. The primary 'win' from using this attribute is if you later drop the reference altogether and folks update to a later package, they won't have source breaks from the dropped reference if they had started actually *using* the library directly.

That said, I suspect you only need this reference to your SequenceOfT.cs file, and it would be relatively trivial for you to rewrite the few lines that need it to just validate arguments directly if you want to drop this reference altogether.

Reply:
+1 to "rewrite the few lines inside SequenceOfT and completely drop the reference"

Reply:
done